protocols/virtualdefund/virtualdefund.go

```diff
@@ -485,48 +498,52 @@ func (o *Objective) validateSignature(sig state.Signature, participantIndex uint
 
 // Update receives an protocols.ObjectiveEvent, applies all applicable event data to the VirtualDefundObjective,
 // and returns the updated state.
-func (o *Objective) Update(event protocols.ObjectiveEvent) (protocols.Objective, error) {
-	if o.Id() != event.ObjectiveId {
-		return o, fmt.Errorf("event and objective Ids do not match: %s and %s respectively", string(event.ObjectiveId), string(o.Id()))
+func (o *Objective) Update(op protocols.ObjectivePayload) (protocols.Objective, error) {
+	if o.Id() != op.ObjectiveId {
+		return o, fmt.Errorf("event and objective Ids do not match: %s and %s respectively", string(op.ObjectiveId), string(o.Id()))
 	}
 
-	updated := o.clone()
+	switch op.Type {
+	case SignedStatePayload:
 
-	if ss := event.SignedState; len(ss.Signatures()) != 0 {
-		incomingChannelId := ss.State().ChannelId()
-		vChannelId := updated.VFixed.ChannelId()
+		updated := o.clone()
 
-		if incomingChannelId != vChannelId {
-			return o, errors.New("event channelId out of scope of objective")
-		} else {
-			incomingSignatures := ss.Signatures()
-			for i := uint(0); i < 3; i++ {
-				existingSig := o.Signatures[i]
-				incomingSig := incomingSignatures[i]
+		incomingSignatures := getSignedStatePayload(op.PayloadData).Signatures()
+		for i := uint(0); i < 3; i++ {
+			existingSig := o.Signatures[i]
+			incomingSig := incomingSignatures[i]
 
-				// If the incoming signature is zeroed we ignore it
-				if isZero(incomingSig) {
+			// If the incoming signature is zeroed we ignore it
+			if isZero(incomingSig) {
+				continue
+			}
+			// If the existing signature is not zeroed we check that it matches the incoming signature
+			if !isZero(existingSig) {
+				if existingSig.Equal(incomingSig) {
 					continue
-				}
-				// If the existing signature is not zeroed we check that it matches the incoming signature
-				if !isZero(existingSig) {
-					if existingSig.Equal(incomingSig) {
-						continue
-					} else {
-						return o, fmt.Errorf("incoming signature %+v does not match existing %+v", incomingSig, existingSig)
-					}
-				}
-				// Otherwise we validate the incoming signature and update our signatures
-				isValid, err := updated.validateSignature(incomingSig, i)
-				if isValid {
-					// Update the signature
-					updated.Signatures[i] = incomingSig
 				} else {
-					return o, fmt.Errorf("failed to validate signature: %w", err)
+					return o, fmt.Errorf("incoming signature %+v does not match existing %+v", incomingSig, existingSig)
 				}
 			}
+			// Otherwise we validate the incoming signature and update our signatures
+			isValid, err := updated.validateSignature(incomingSig, i)
+			if isValid {
+				// Update the signature
+				updated.Signatures[i] = incomingSig
+			} else {
+				return o, fmt.Errorf("failed to validate signature: %w", err)
+			}
 		}
+		return &updated, nil
+	case InitPayload:
+		panic("TODO: Not implemented yet")
```

Comment:
Just return an error instead of panicking

Reply:
https://github.com/statechannels/go-nitro/pull/864/commits/2ea8ab2a42b5433630aaed22ca3762bbaaa556ef